packages/components/src/switch/src/Switch.css

```diff
@@ -42,103 +54,158 @@
+.o-ui-switch-focus:not(.o-ui-switch-disabled) .o-ui-switch-control {
+    background-color: var(--o-ui-switch-focus-background);
+    border-color: var(--o-ui-switch-focus-border);
+    box-shadow: 0 0 0 0.12rem var(--hop-neutral-surface), 0 0 0 0.219rem var(--hop-primary-border-focus);
 }
 
-/* STATE */
-/* STATE | CHECKED */
-.o-ui-switch-checked .o-ui-switch-control {
-    background-color: var(--o-ui-bg-alias-accent);
+.o-ui-switch input[type="checkbox"]:focus-visible + .o-ui-switch-control::before,
+.o-ui-switch-focus:not(.o-ui-switch-disabled) .o-ui-switch-control::before {
+    background-color: var(--o-ui-switch-focus-icon);
 }
 
-.o-ui-switch-checked .o-ui-switch-control::before {
-    transform: translate(calc(var(--o-ui-sz-6) / 2 + 2px), 2px);
+/* STATE | CHECKED */
+.o-ui-switch-checked .o-ui-switch-control,
+.o-ui-switch-checked:not(.o-ui-switch-disabled):hover .o-ui-switch-control,
+.o-ui-switch-checked.o-ui-switch-hover:not(.o-ui-switch-disabled) .o-ui-switch-control,
+.o-ui-switch-checked input[type="checkbox"]:focus-visible + .o-ui-switch-control,
+.o-ui-switch-checked.o-ui-switch-focus:not(.o-ui-switch-disabled) .o-ui-switch-control {
+    background-color: var(--hop-neutral-surface-active);
+    border-color: var(--hop-neutral-border-active);
 }
 
-/* STATE | FOCUS */
-.o-ui-switch-control:after {
-    opacity: 0;
-    content: "";
-    top: 0;
-    left: 0;
-    right: 0;
-    bottom: 0;
-    display: block;
-    position: absolute;
-    box-shadow: 0 0 0 var(--o-ui-focus-ring-thickness-md) var(--o-ui-focus-ring-color-alias-default);
-    border-radius: var(--o-ui-br-pill);
-    transition: var(--o-ui-focus-ring-transition);
+.o-ui-switch-checked .o-ui-switch-control::before {
+    transform: translate(calc(3rem / 2 + var(--o-ui-switch-control-before-transform-translate-value)), var(--o-ui-switch-control-before-transform-translate-value));
 }
 
-.o-ui-switch input[type="checkbox"]:focus-visible + .o-ui-switch-control:after,
-.o-ui-switch-focus:not(.o-ui-switch-disabled) .o-ui-switch-control:after {
-    opacity: 1;
+.o-ui-switch-checked .o-ui-switch-control::before,
+.o-ui-switch-checked:not(.o-ui-switch-disabled):hover .o-ui-switch-control::before,
+.o-ui-switch-checked.o-ui-switch-hover:not(.o-ui-switch-disabled) .o-ui-switch-control::before,
+.o-ui-switch-checked input[type="checkbox"]:focus-visible + .o-ui-switch-control::before,
+.o-ui-switch-checked.o-ui-switch-focus:not(.o-ui-switch-disabled) .o-ui-switch-control::before {
+    background-color: var(--hop-neutral-icon-active);
 }
 
 /* STATE | DISABLED */
-.o-ui-switch[disabled],
 .o-ui-switch-disabled {
-    opacity: var(--o-ui-disabled-opacity);
     cursor: not-allowed;
 }
 
+.o-ui-switch-disabled .o-ui-switch-control {
+    background-color: var(--hop-neutral-surface-disabled);
+    border-color: var(--hop-neutral-border-disabled);
+}
+
+.o-ui-switch-disabled .o-ui-switch-control::before {
+    background-color: var(--hop-neutral-icon-disabled);
+}
+
+/* INVALID */
```

Comment:
The design has no notion of invalid, I don't see how a toggle could be in error. I would remove this until we hear back from design, it they need it, we could do another PR to add this.